src/posix/platform/infra_if.cpp

```diff
@@ -708,6 +723,122 @@ otError InfraNetif::DiscoverNat64Prefix(uint32_t aInfraIfIndex)
 }
 #endif // OPENTHREAD_POSIX_CONFIG_NAT64_AIL_PREFIX_ENABLE
 
+#if OPENTHREAD_POSIX_CONFIG_TRACK_ADDRESSES_ON_ANY_NETIF
+bool InfraNetif::AnyNetifHasAddress(const otIp6Address &aAddress)
```

Comment:
I don't think this should be in the `InfraNetif` class even though there is already code which is using "getifaddrs()". `InfraNetif` should for just things related to the AIL, this platform API `otPlatHostHasAddress` is not.

if you don't find a good alternative, just create a new misc.hpp and misc.cpp for it.